crates/polars-core/src/frame/mod.rs

```diff
@@ -3255,6 +3271,7 @@ impl DataFrame {
 
 pub struct RecordBatchIter<'a> {
     columns: &'a Vec<Column>,
+    schema: ArrowSchemaRef,
```

Comment:
Why do we want the schema on `RecordBatch`? Do we want O(1) access somewhere?

Reply:
We need it to be able to properly round trip types. We need the ArrowFields, the ArrowDataType isn't enough.

Reply:
Ah right. Convinces me more that we should shift to `Fields` on `Series`/`Columns`.